tool/tsh/common/app.go

```diff
@@ -617,23 +622,38 @@ func (a *appInfo) GetApp(ctx context.Context, tc *client.TeleportClient) (types.
 }
 
 // getApp returns the registered application with the specified name.
-func getApp(ctx context.Context, tc *client.TeleportClient, name string) (app types.Application, err error) {
-	var apps []types.Application
-	err = client.RetryWithRelogin(ctx, tc, func() error {
-		apps, err = tc.ListApps(ctx, &proto.ListResourcesRequest{
-			Namespace:           tc.Namespace,
+func getApp(ctx context.Context, tc *client.TeleportClient, clt apiclient.GetResourcesClient, name string) (app types.Application, logins []string, err error) {
+	var res apiclient.ResourcePage[*types.EnrichedResource]
+	// Fetching the app from the auth server needs to be done on a
+	// RetryWithRelogin block for cases where the certificates expired or a
+	// login is required (e.g., configuration change). We don't need a new
+	// instance of GetResourcesClient since it will reconnect using credentials
+	// from the updated TeleportClient.
+	if err := client.RetryWithRelogin(ctx, tc, func() error {
+		// When listing a single app we only need to grab one page.
+		res, err = apiclient.GetEnrichedResourcePage(ctx, clt, &proto.ListResourcesRequest{
 			ResourceType:        types.KindAppServer,
+			SortBy:              types.SortBy{Field: types.ResourceMetadataName},
 			PredicateExpression: fmt.Sprintf(`name == "%s"`, name),
+			Limit:               1,
```

Comment:
Is the `PredicateExpression` by the backend or is it used as a filter after the backend ranges?

If later, is limit 1 used in each backend call?